### DISCUSSION TITLE
Agents Experimental Feature Not Working (Asking for key)

### DISCUSSION THREAD

**Discussion by MikeNatC** [danny-avila/LibreChat#4886] [Help Wanted]
I have enable the Agents, experimental feature following this discussion: https://github.com/danny-avila/LibreChat/issues/3607#issuecomment-2395089567

I am able to get the Agent custom endpoint. However, when I test a basic agent (without Tools or Actions) that I created using the agent builder (i.e. Model 4o-mini), I get the error message: "No key found. Please provide a key and try again."

I'm not sure what I've done wrong and would appreciate some help!

**Comment by jiujianzx** [comment]
same qs

**Reply by MikeNatC** [reply]
This was moved to a discussion [here](https://github.com/danny-avila/LibreChat/discussions/4902) and resolved via changes to the code by Danny. 

I took down the LibreChat container, pulled the latest updates and then restarted the container with a build command. This solved my problem.

**Reply by danny-avila** [reply]
@jiujianzx make sure your `user_provided` key is provided via the regular endpoint option
